### PR TITLE
Support for `data_masking` in `azurerm_api_management_api_diagnostic`

### PR DESCRIPTION
**Pull request by yupwei68**
Fix https://github.com/terraform-providers/terraform-provider-azurerm/issues/12418

--- PASS: TestAccApiManagementApiDiagnostic_basic (292.38s)
--- PASS: TestAccApiManagementApiDiagnostic_update (404.77s)
--- PASS: TestAccApiManagementApiDiagnostic_requiresImport (312.74s)
--- PASS: TestAccApiManagementApiDiagnostic_complete (290.99s)
--- PASS: TestAccApiManagementApiDiagnostic_dataMasking (377.64s)

### CODE REVIEW

File: azurerm/internal/services/apimanagement/api_management_api_diagnostic_resource.go [hunk 1]
```diff
@@ -135,6 +135,7 @@ func resourceApiManagementApiDiagnosticAdditionalContentSchema() *pluginsdk.Sche
 					},
 					Set: pluginsdk.HashString,
 				},
+				"data_masking": schemaz.SchemaApiManagementDataMasking(),
```
Comment (katbyte) [inline azurerm/internal/services/apimanagement/api_management_api_diagnostic_resource.go:138]: how come we are not inlining this and pulling it out into a function?

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by yupwei68** [comment]
kt, Please continue reviewing.

File: website/docs/r/api_management_api_diagnostic.html.markdown [hunk 2]
```diff
@@ -155,6 +155,24 @@ A `backend_request`, `backend_response`, `frontend_request` or `frontend_respons
...
+
+A `data_masking` block supports the following:
+
+* `query_params` - (Optional) A `query_params` block as defined below.
+
+* `headers` - (Optional) A `headers` block as defined below.
+
+---
+
+A `query_params`, `headers` block supports the following:
```
Comment (katbyte) [inline website/docs/r/api_management_api_diagnostic.html.markdown:170]:
```suggestion
The `query_params` and `headers` blocks support the following:
```

**Review by katbyte — APPROVED** [review]
aside from a couple minor comments this LGTM 🍔

**Pull request merged**